nixos/modules/services/backup/duplicity-backup.nix

```diff
@@ -0,0 +1,174 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.duplicityBackup;
+
+  cfgPackage = pkgs.duplicity-backup-sh;
+
+  duplicityBackupConfig = cfg: ''
+    ROOT="${cfg.root}"
+    DEST="${cfg.destination}"
+    ${optionalString (cfg.include != []) ''INCLIST=(
+      ${concatMapStringsSep " " (x: ''"${x}"'') cfg.include}
+    )''}
+    ${optionalString (cfg.exclude != []) ''EXCLIST=(
+      ${concatMapStringsSep " " (x: ''"${x}"'') cfg.exclude}
+    )''}
+    ENCRYPTION="${if cfg.encryption.enable
+      then "yes" else "no"
+    }"
+    ${optionalString cfg.encryption.enable ''
+      PASSPHRASE="${cfg.encryption.passphrase}"
+    ''}
+    CLEAN_UP_TYPE="${if cfg.cleanup.enable then cfg.cleanup.type else "none"}"
+    ${optionalString cfg.cleanup.enable '' 
+      CLEAN_UP_VARIABLE="${cfg.cleanup.var}"
+    ''}
+    HOSTNAME=${config.networking.hostName}
+    STATIC_OPTIONS="${cfg.options}"
+    LOGDIR="/var/log/duplicity-backup/"
+    LOG_FILE="duplicity-$(date +%Y-%m-%d_%H-%M).txt"
+    VERBOSITY="-v3"
+    ${cfg.extraConfig}
+  '';
+
+in {
+
+  options = {
+    services.duplicityBackup = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Duplicity.
+        '';
+      };
+
+      onCalendar = mkOption {
+        type = types.str;
+        default = "02:00";
+        description = ''
+          Create backup on given date/time/interval, which must be in the
+          format described in
+          <citerefentry><refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum></citerefentry>.
+        '';
+      };
+
+      options = mkOption {
+        type = types.str;
+        default = "";
+        description = "Options to run duplicity with.";
+      };
+
+      extraConfig = mkOption {
+        type = types.str;
```

Comment:
Consider `types.lines` here, I think it has more intuitive merge semantics for an option like this.